tests/test_client_request.py

```diff
@@ -1230,3 +1232,47 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "trust_env", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            True,
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            False,
+            None,
+        ),
+        ("", "example.com", True, None),
+        ("", "example.com", False, None),
+    ],
+)
+def test_basicauth_from_netrc(
+    make_request: Any,
+    netrc_contents: str,
+    hostname: str,
+    trust_env: bool,
+    expected_auth: Optional[helpers.BasicAuth],
+):
+    with tempfile.NamedTemporaryFile() as f:
+        f.write(netrc_contents.encode())
+        f.flush()
+        try:
+            # save and restore NETRC env variable if already set
```

Comment:
Instead of this, use the built-in `monkeypatch` fixture that will unpatch the env automatically after the test function finishes.

Reply:
TIL about https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.setenv, which is super useful! Thanks for pointing it out :)